clang/lib/AST/DeclBase.cpp

```diff
@@ -1428,6 +1428,17 @@ bool DeclContext::Encloses(const DeclContext *DC) const {
   return false;
 }
 
+bool DeclContext::LexicallyEncloses(const DeclContext *DC) const {
+  if (getPrimaryContext() != this)
+    return getPrimaryContext()->LexicallyEncloses(DC);
+
+  for (; DC; DC = DC->getLexicalParent())
+    if (!isa<LinkageSpecDecl>(DC) && !isa<ExportDecl>(DC) &&
```

Comment:
```suggestion
    if (!isa<LinkageSpecDecl, ExportDecl>(DC) &&
```

That said... Perhaps we should just do a 

```
if (isa<LinkageSpecDecl, ExportDecl>(*this)) return false;
```
??  Since thats the only way the equivilence would work?

Reply:
This code was lifted from `DeclContext::Encloses()`, so the cleanup is easy because it doesn't change logic between the two functions. I'll clean that up, but other changes should be done in a follow-up so they're done consistently (if done at all). WDYT?

Reply:
Makes sense to me, I'm fine with the rest of it in a followup.